wiring/inc/spark_wiring_ble.h

```diff
@@ -197,52 +183,83 @@ class BleCharacteristicHandles : public hal_ble_char_handles_t {
 static_assert(std::is_pod<BleCharacteristicHandles>::value, "BleCharacteristicHandles is not a POD struct");
 
 
+class BleAddress {
+public:
+    BleAddress();
+    BleAddress(const hal_ble_addr_t& addr);
+    BleAddress(const uint8_t addr[BLE_SIG_ADDR_LEN], BleAddressType type = BleAddressType::PUBLIC);
+    BleAddress(const char* address, BleAddressType type = BleAddressType::PUBLIC);
+    BleAddress(const String& address, BleAddressType type = BleAddressType::PUBLIC);
+    ~BleAddress() = default;
+
+    // Setters
+    int type(BleAddressType type);
+    int set(const uint8_t addr[BLE_SIG_ADDR_LEN], BleAddressType type = BleAddressType::PUBLIC);
+    int set(const char* address, BleAddressType type = BleAddressType::PUBLIC);
+    int set(const String& address, BleAddressType type = BleAddressType::PUBLIC);
+    BleAddress& operator=(const hal_ble_addr_t& addr);
+    BleAddress& operator=(const uint8_t addr[BLE_SIG_ADDR_LEN]);
+
+    // Getters
+    BleAddressType type() const;
+    void octets(uint8_t addr[BLE_SIG_ADDR_LEN]) const;
+    String toString(bool stripped = false) const;
+    size_t toString(char* buf, size_t len, bool stripped = false) const;
+    hal_ble_addr_t halAddress() const;
+    uint8_t operator[](uint8_t i) const;
+
+    bool operator==(const BleAddress& addr) const;
+
+private:
+    void toBigEndian(uint8_t buf[BLE_SIG_ADDR_LEN]) const;
+
+    hal_ble_addr_t address_;
+};
+
+
 class BleUuid {
 public:
-    BleUuid()
-            : uuid_() {
-    }
+    BleUuid();
+    BleUuid(const hal_ble_uuid_t& uuid);
     BleUuid(const BleUuid& uuid);
     BleUuid(const uint8_t* uuid128, BleUuidOrder order = BleUuidOrder::LSB);
-    BleUuid(uint16_t uuid16);
     BleUuid(const uint8_t* uuid128, uint16_t uuid16, BleUuidOrder order = BleUuidOrder::LSB);
-    BleUuid(const String& uuid);
-    BleUuid(const char* uuid);
+    explicit BleUuid(uint16_t uuid16);
```

Comment:
I mark these constructors as `explicit` so that it won't allow to convert a string or uint16_t value to a UUID object by calling `getCharacteristicByUUID(BleCharacteristic* characteristic, const BleUuid& uuid)`, instead it should call the `getCharacteristicByUUID(BleCharacteristic* characteristic, T uuid)` for better understanding. While the `BleAddress` doesn't appear to have this kind of worry.

Reply:
But that results in pretty much the same behavior, right?

Reply:
Yes. it just simplifies the understanding of which prototype is referenced at the first glance of the wiring APIs.

Reply:
But in that case the templated method becomes unnecessary. I'd say we either allow implicit conversions for both `BleUuid` and `BleAddress`, or we disallow for both.

Reply:
I'll remove the `explicit` for `BleUuid` and the template method getting characteristics by UUID.